group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -720,19 +764,121 @@ private CoordinatorResult<ConsumerGroupHeartbeatResponseData, Record> consumerGr
         );
 
         if (!subscriptionMetadata.equals(group.subscriptionMetadata())) {
-            log.info("[GroupId " + groupId + "] Computed new subscription metadata: "
+            log.info("[GroupId " + group.groupId() + "] Computed new subscription metadata: "
                 + subscriptionMetadata + ".");
-            records.add(newGroupSubscriptionMetadataRecord(groupId, subscriptionMetadata));
+            records.add(newGroupSubscriptionMetadataRecord(group.groupId(), subscriptionMetadata));
         }
 
         // We bump the group epoch.
         int groupEpoch = group.groupEpoch() + 1;
```

Comment:
so that i understand, the group epoch will be incremented when we replay the records that were committed to the log.

after an epoch bump, what happens to the timeouts, are all members fenced or do they not rely on the group epoch?

Reply:
They don't rely on the group epoch. Members are only fenced by on their member epoch.